lib/chef/resource/screensaver.rb

```diff
@@ -0,0 +1,175 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../resource"
+
+class Chef
+  class Resource
+    class Screensaver < Chef::Resource
+      resource_name :screensaver
+      provides(:screensaver) { true }
+
+      description "Use the screensaver resource to configure screensaver settings."
+      introduced "15.1"
+
+      property :idle_time, Integer,
+               description: "Number of seconds before screensaver activates."
+
+      property :ask_for_password, [TrueClass, FalseClass],
+               description: "Ask for password to bypass screensaver."
+
+      property :password_delay, Integer,
+               description: "How many seconds to wait before asking for password."
+
+      property :organization, String,
+               description: "How many seconds to wait before asking for password.",
+               default: "Chef"
+
+      property :prefix , String,
+               description: "How many seconds to wait before asking for password.",
+               default: "com.chef"
+
+      action :manage do
+        description "Manage screensaver settings."
+
+        converge_by("Manage #{@new_resource}") do
+          case node[:platform_family]
+          when "mac_os_x"
+            manage_screensaver_osx
+          when "windows"
+            manage_screensaver_win
+          end
+        end
+      end
+
+      action :unmanage do
+        description "Unmanage screensaver settings"
+
+        converge_by("Unmanage #{@new_resource}") do
+          case node[:platform_family]
+          when "mac_os_x"
+            unmanage_screensaver_osx
+          when "windows"
+            unmanage_screensaver_win
+          end
+        end
+      end
+
+      action_class do
+        def profile_identifier
+          "#{new_resource.prefix}.chef.screensaver"
+        end
+
+        def manage_screensaver_osx
+          screensaver_profile = {
+            'PayloadIdentifier' => profile_identifier,
+            'PayloadRemovalDisallowed' => true,
+            'PayloadScope' => 'System',
+            'PayloadType' => 'Configuration',
+            'PayloadUUID' => 'CEA1E58D-9D0F-453A-AA52-830986A8366C',
+            'PayloadOrganization' => new_resource.organization,
+            'PayloadVersion' => 1,
+            'PayloadDisplayName' => 'Screensaver',
+            'PayloadContent' => []
+          }
+          screensaver_payload = {
+            'PayloadType' => 'com.apple.screensaver',
+            'PayloadVersion' => 1,
+            'PayloadIdentifier' => profile_identifier,
+            'PayloadUUID' => '3B2AD6A9-F99E-4813-980A-4147617B2E75',
+            'PayloadEnabled' => true,
+            'PayloadDisplayName' => 'Screensaver'
+          }
+
+          screensaver_payload['idleTime'] = new_resource.idle_time if new_resource.idle_time
```

Comment:
instead of doing this, should any value be allowed, in case Apple adds new keys later?

Reply:
We would need to add a property to the resource to consume the new key, which means touching the code anyways. For a core resource, I think it is better to keep it tightly scoped and update as needed.